### PR TITLE
Upgrade maven compiler plugin, bouncycastle e2e/sample dependency versions

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Jackson upgrades as well

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -142,13 +142,18 @@
             </dependency>
             <dependency>
                 <groupId>org.bouncycastle</groupId>
-                <artifactId>bcmail-jdk15on</artifactId>
-                <version>1.70</version>  <!--e2e tests and samples only-->
+                <artifactId>bcmail-jdk18on</artifactId>
```
Comment (timtay-microsoft) [inline pom.xml:145]: jdk15on versions are no longer maintained, so we need to move to jdk18on to stay up to date on security patches

File: pom.xml [hunk 2]
```diff
@@ -142,13 +142,18 @@
...
             <dependency>
                 <groupId>org.bouncycastle</groupId>
-                <artifactId>bcprov-jdk15on</artifactId>
-                <version>1.70</version>  <!--e2e tests and samples only-->
+                <artifactId>bcprov-jdk18on</artifactId>
+                <version>1.77</version>  <!--e2e tests and samples only-->
+            </dependency>
+            <dependency>
+                <groupId>org.bouncycastle</groupId>
+                <artifactId>bcpkix-jdk18on</artifactId>
```
Comment (timtay-microsoft) [inline pom.xml:155]: This new dependency is a new part of the "jdk18on" set of bouncycastle packages

**Review by brycewang-microsoft — APPROVED** [review]

**Comment by timtay-microsoft** [comment]
We still get the old bouncycastle dependency from our TSS library, so I'll close this for now while we work out what we want to do about that dependency

**Pull request closed without merging**